cmds/core/touch/touch.go

```diff
@@ -0,0 +1,98 @@
+// Copyright 2023 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+package main
+
+import (
+	"flag"
+	"log"
+	"os"
+	"time"
+)
+
+type params struct {
+	time         time.Time
+	access       bool
+	modification bool
+	create       bool
+}
+
+type cmd struct {
+	params
+	args []string
+}
+
+func command(p params, args ...string) *cmd {
+	return &cmd{
+		args:   args,
+		params: p,
+	}
+}
+
+func parseParams(dateTime string, access, modification, create bool) (params, error) {
+	flag.Parse()
+	t := time.Now()
+	if dateTime != "" {
+		var err error
+		t, err = time.Parse(time.RFC3339, dateTime)
+		if err != nil {
+			return params{}, err
+		}
+	}
+	return params{
+		access:       access || (!access && !modification),
+		modification: modification || (!access && !modification),
+		create:       create,
+		time:         t,
+	}, nil
+}
+
+func (c *cmd) run() error {
+	for _, arg := range c.args {
+		_, existsErr := os.Stat(arg)
+		notExist := os.IsNotExist(existsErr)
+		if notExist {
+			if c.create {
+				continue
+			}
+
+			f, err := os.Create(arg)
+			if err != nil {
+				return err
+			}
+			f.Close()
+		}
+
+		accessTime := time.Time{}
+		if c.access || notExist {
+			accessTime = c.time
+		}
+		modificationTime := time.Time{}
+		if c.modification || notExist {
+			modificationTime = c.time
+		}
+
+		err := os.Chtimes(arg, accessTime, modificationTime)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func main() {
+	var access = flag.Bool("a", false, "change only the access time")
```

Comment:
also, it's up to you, but do you want to do the declaration as
var (
...
)

styly or do you feel the one var per line is nicer looking. Also, might you do
access :=
instead of var access = instead?